src/drivers/drivers.jl

```diff
@@ -85,9 +85,31 @@ function set_driver!(p::Properties, ::POSIX)
 end
 
 function __init__()
-    DRIVERS[API.h5fd_core_init()] = Core
-    DRIVERS[API.h5fd_sec2_init()] = POSIX
-    @require MPI="da04e1cc-30fd-572f-bb4f-1f8673147195" include("mpio.jl")
+    # disable file locking as that can cause problems with mmap'ing
+    if !haskey(ENV, "HDF5_USE_FILE_LOCKING")
+        ENV["HDF5_USE_FILE_LOCKING"] = "FALSE"
+    end
+
+    fapl = HDF5.init!(HDF5.FileAccessProperties())
+    API.h5p_set_fapl_core(fapl, 4096, false)
+    DRIVERS[API.h5p_get_driver(fapl)] = Core
+    API.h5p_set_fapl_sec2(fapl)
+    DRIVERS[API.h5p_get_driver(fapl)] = POSIX
+    close(fapl)
+
+    # Check whether the loaded HDF5 libraries were compiled with parallel support.
+    loaded_libs = filter(s->match(r"hdf5"i, s) !== nothing, Libc.Libdl.dllist())
+    for libname in loaded_libs
+        has_mpio = Libc.Libdl.dlopen(libname) do lib
+            Libc.Libdl.dlsym(lib, :H5Pset_fapl_mpio; throw_error=false) !== nothing
+        end
+        if has_mpio
+            HDF5.HAS_PARALLEL[] = true
+            break
+        end
+    end
```

Comment:
```suggestion
    HDF5.HAS_PARALLEL[] = Libdl.dlsym(API.HDF5_jll.libhdf5_handle, :H5Pset_fapl_mpio; throw_error=false) !== nothing
```

Add `using Libdl` above.

Reply:
I don't think this will work with a custom HDF5 library either